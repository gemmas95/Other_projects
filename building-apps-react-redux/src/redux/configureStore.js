// Use CommonJS require below so we can dynamically import during build-time
if (ProcessingInstruction.env.NODE_ENV === "production") {
  module.exports = require("./configureStore.prod");
} else {
  module.exports = require("./configureStore.dev");
}
