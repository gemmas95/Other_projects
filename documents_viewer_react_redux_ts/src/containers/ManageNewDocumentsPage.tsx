import React, { useState } from "react";
import { connect } from "react-redux";
import { useHistory } from "react-router-dom";
import { toast } from "react-toastify";
import DocumentForm from "../components/documents/DocumentForm";

// Actions
import { createDocument } from "../redux/actions/documentActions";

// Interfaces and types
import { ThunkDispatch } from "redux-thunk";
import { AppActions } from "../redux/actions/actionTypes";
import IDocument from "../models/document.interface";

interface Props {
  saveDocument: Function;
}

interface IError {
  message: string;
  error: string;
}

const ManageNewDocumentsPage: React.FC<Props> = ({ saveDocument }) => {
  const history = useHistory();
  const [newData, setNewData] = useState({
    id: Math.ceil(Math.random() * 10000000000) + "",
    title: "",
    type: "",
    date: "",
    text: "",
    image: "",
  });
  const [saving, setSaving] = useState<boolean>(false);

  function handleSubmit(event: React.ChangeEvent<HTMLInputElement>) {
    event.preventDefault();
    setSaving(true);
    saveDocument(newData)
      .then(() => {
        toast.success("Document saved!");
        history.push("/documents");
      })
      .catch((error: IError) => {
        setSaving(false);
        console.error(error);
        toast.error("Oops, something went wrong...", { autoClose: false });
      });
  }

  function handleChange(event: any) {
    event.preventDefault();
    const { name, value } = event.target;
    setNewData({ ...newData, [name]: value });
  }

  return (
    <>
      <h1>Add Document</h1>
      <DocumentForm
        handleSubmit={handleSubmit}
        handleChange={handleChange}
        saving={saving}
      />
    </>
  );
};

function mapDispatchToProps(dispatch: ThunkDispatch<any, any, AppActions>) {
  return {
    saveDocument: (newData: IDocument) => dispatch(createDocument(newData)),
  };
}

export default connect(null, mapDispatchToProps)(ManageNewDocumentsPage);
