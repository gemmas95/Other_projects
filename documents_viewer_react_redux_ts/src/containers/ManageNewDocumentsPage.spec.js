import React from "react";
import ManageNewDocumentsPage from "./ManageNewDocumentsPage";

import { BrowserRouter } from "react-router-dom";

import renderer from "react-test-renderer";

import { Provider } from "react-redux";
import configureStore from "redux-mock-store";

// Actions
// import { saveDocument } from "../redux/actions/documentActions";

const mockStore = configureStore([]);

describe("ManageNewDocumentsPage", () => {
  let store;

  beforeEach(() => {
    store = mockStore({
      documents: {
        data: [{}],
        visibleDocuments: [{}],
      },
    });

    store.dispatch = jest.fn();
  });
  it("should match snapshot", () => {
    const ManageNewDocumentsPageTree = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <ManageNewDocumentsPage />
        </BrowserRouter>{" "}
      </Provider>
    );
    expect(ManageNewDocumentsPageTree.toJSON()).toMatchSnapshot();
  });
});
