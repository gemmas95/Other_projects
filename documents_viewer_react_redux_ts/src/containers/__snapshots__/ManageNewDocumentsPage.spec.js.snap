// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ManageNewDocumentsPage should match snapshot 1`] = `
Array [
  <h1>
    Add Document
  </h1>,
  <form
    autoComplete="off"
    onSubmit={[Function]}
  >
    <p>
      If you choose 
      <strong>
        simple type
      </strong>
       of document even though you select an image or text, it will only be displayed the title and date.
       
      <br />
      For 
      <strong>
        custom type
      </strong>
       will be displayed text too. 
      <br />
      And for the 
      <strong>
        advanced type
      </strong>
       all items will be displayed.
    </p>
    <label
      htmlFor="title"
    >
      Title
    </label>
    <input
      id="title"
      maxLength={80}
      name="title"
      onChange={[Function]}
      required={true}
      type="text"
    />
    <label
      htmlFor="type"
    >
      Type
    </label>
    <select
      id="type"
      name="type"
      onChange={[Function]}
      required={true}
    >
      <option
        value=""
      >
        Choose a type
      </option>
      <option
        value="simple"
      >
        Simple
      </option>
      <option
        value="custom"
      >
        Custom
      </option>
      <option
        value="advanced"
      >
        Advanced
      </option>
    </select>
    <label
      htmlFor="date"
    >
      Date
    </label>
    <input
      id="date"
      name="date"
      onChange={[Function]}
      required={true}
      type="date"
    />
    <label
      htmlFor="text"
    >
      Text
    </label>
    <textarea
      id="text"
      name="text"
      onChange={[Function]}
    />
    <label
      htmlFor="image"
    >
      Image
    </label>
    <input
      id="image"
      name="image"
      onChange={[Function]}
      type="text"
    />
    <p>
      The format of the image must be an url ending with a proper image extension(p.e: .jpeg, .jpg, gif,...) 
      <br />
      <span>
        (example: https://mimusica.es/wp-content/uploads/2019/10/musica-chillout.jpg)
      </span>
    </p>
    <button
      className="greenButton"
      disabled={false}
      type="submit"
    >
      Create course
    </button>
  </form>,
  " ",
]
`;
